config/main.py

```diff
@@ -2656,6 +2675,8 @@ def add(ctx, interface_name, ip_addr, gw):
         if '/' not in ip_addr:
             ip_addr = str(net)
 
+        ip_addr = validate_ip_address(ctx, ip_addr)
```

Comment:
Not sure if you want to call this "validate". Since this is for IPv6, why dont we just call if address is v6 and update the leading zeros?

Reply:
This function is not only for IPv6, it tries to detect the IP type and validate it. According to the [ipaddress doc](https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Address), IPv4Address creates an IP address and optionally removes extra zeros if the value is valid, but returns nothing other than an exception if the IP is invalid. Therefore, this is the only way to check the value with this function.

Reply:
This feature is not only for IPv6, it checks the range of digits for v4 and v6 IP addresses and additionally removes leading zeros.

Reply:
@d-dashkov it seems to me the validation is only on the mast for ipv4 and ipv6. in this case we can do the following
- rename the function to validate_ip_mask
- add more validation points so it can be generic function.

Reply:
I refactored the code, and now all IP checking has been moved to my separate function and the try-except construct has been removed from the add and remove ip interface functions. So now it is a generic function that returns the IP string without leading zeros, or False if something is wrong.